enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1186 @@
+attention.
+
+#### Change Management on Master MachineConfigPool
+In order to allow control-plane updates as a single material change, the MCO will only honor change the management configuration for the 
+master MachineConfigPool if user generated MachineConfigs are the cause for a pending change. To accomplish this, 
+at least one MachineConfig updated by the CVO will have the `machineconfiguration.openshift.io/bypass-change-management` annotation
+indicating that changes in the MachineConfig must be acted upon irrespective of the master MCP change management policy. 
+
+#### Limiting Overlapping Window Search / Permissive Window Calculation
+An operator implementing change management for a worker-node related resource must honor the change management hierarchy when
+calculating when the next permissive window will occur (called elsewhere in the document, ETA). This is not
+straightforward to compute when both the control-plane and worker-nodes have independent MaintenanceSchedule 
+configurations.
+
+We can, however, simplify the process by reducing the number of days in the future the algorithm must search for
+coinciding permissive windows. 1000 days is a proposed cut-off.
+
+To calculate coinciding windows then, the implementation can use [rrule-go](https://github.com/teambition/rrule-go) 
+to iteratively find permissive windows at the cluster / control-plane level. These can be added to an 
+[interval-tree](https://github.com/rdleal/intervalst) . As dates are added, rrule calculations for the worker-nodes
+can be performed. The interval-tree should be able to efficiently determine whether there is an
+intersection between the permissive intervals it has stored for the control-plane and the time range tested for the 
+worker-nodes.
+
+Since it is possible there is no overlap, limits must be placed on this search. Once dates >1000 days from
+the present moment are being tested, the operator can behave as if the next window will occur in
+1000 days (prevents infinite search for overlap).
+
+This outcome does not need to be recomputed unless the operator restarts Or one of the RRULE involved
+is modified.
+
+If an overlap _is_ found, no additional intervals need to be added to the tree and it can be discarded.
+The operator can store the start & end datetimes for the overlap and count down the seconds remaining 
+until it occurs. Obviously, this calculation must be repeated:
+1. If either MaintenanceSchedule configuration is updated.
+1. The operator is restarted.
+1. At the end of a permissive window, in order to determine the next permissive window.
+
+
+#### Service Delivery Option Sanitization
+It is obvious that the range of flexible options provided by change management configurations offers
+can create risks for inexperienced cluster lifecycle administrators. For example, setting a 
+standalone cluster to use the Assisted strategy and failing to trigger worker-node updates will

```

Comment:
Managed OpenShift clusters are not able to change the MachineConfigPool strategy due to platform restrictions. We don't expect this to be a supported configuration.

Reply:
+1 - just an example of why restrictions for a managed fleet are a good idea.